### PR TITLE
Mentor email registation

### PR DESCRIPTION
**Pull request by wley3337**
Adds a mentor signup form -> generate email ( email is not sent but that's the only step missing) You can use the link printed out by the backend to take you to the login form. Not sure where we want to direct the user from there or what decisions we want to use to move them to the next page

### CODE REVIEW

**Comment by wley3337** [comment]
I don't know that all of these commits really are part of this. i think it's because I branched off of branch 🤦

File: src/redux/registration/registration.actions.ts [hunk 1]
```diff
@@ -85,3 +92,52 @@ function* hanldeSubmitStudentRegistration(
...
+export function* watchSubmitMentorSignUpForm() {
+  yield takeLatest(SUBMIT_MENTOR_SIGN_UP, handleSubmitMentorSignUpForm);
+}
+function* handleSubmitMentorSignUpForm(action: SubmitMentorSignUpAction) {
+  const defaultErrorMessage = 'There was an error processing your sign up.';
+  const current_errors: Errors = yield select(
+    (state: AppState) => state.errors
+  );
+  try {
+    const { mentorSignUpForm, navigateFunction } = action.payload;
```
Comment (geekingreen) [inline src/redux/registration/registration.actions.ts:111]: I haven't tested this, but I think we can avoid passing the `react-router` navigate function to our actions and instead use `import history from 'history/browser';` and use that directly within our actions like `history.push('<url>');`

**Comment by geekingreen** [comment]
looking good, if that `history` import works I think that would make navigation a little easier without needing to pass that function around.

File: src/redux/registration/registration.actions.ts (hunk 1 above)
Comment (wley3337) [reply src/redux/registration/registration.actions.ts:111]: @geekingreen it looks like it updates the url but doesn't seem to actually push the browser to that ref... 🤔

Comment (wley3337) [reply src/redux/registration/registration.actions.ts:111]: This is what I tried. I also tried with a `import {createHistory} from 'history'; const history = createHistory()` as well

```ts
import history from 'history/browser';

const pushHistory = (path: string): void => {
  history.push(path);
};

export default pushHistory;

```

**Comment by wley3337** [comment]
Yeah, it would be nice. I think React-Router isn't able to `see` the change even though they're using the `history` library as well. I wonder if it's just that they're using a different instance of the `history()`

File: src/redux/registration/registration.actions.ts (hunk 1 above)
Comment (geekingreen) [reply src/redux/registration/registration.actions.ts:111]: darn... I was hopeful... well if I think of any other "wise" ideas I'll let you know

Comment (geekingreen) [reply src/redux/registration/registration.actions.ts:111]: Well I did come up with a solution. Totally optional if we want to go this route though:
```tsx
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { ReactNode, useLayoutEffect, useState } from 'react';

export const history = createBrowserHistory();

export const BrowserRouter = ({ children }: { children: ReactNode }) => {
  const [state, setState] = useState({
    action: history.action,
    location: history.location,
  });

  useLayoutEffect(() => {
    const unlisten = history.listen(setState);
    return () => unlisten();
  }, []);

  return (
    <Router
      location={state.location}
      navigator={history}
      navigationType={state.action}
    >
      {children}
    </Router>
  );
};
```

Comment (geekingreen) [reply src/redux/registration/registration.actions.ts:111]: basically a clone of `BrowserRouter` but with our static `history` https://github.com/remix-run/react-router/blob/0f9435a8134b2b5dddfd716a18d17aefe4461fe1/packages/react-router-dom/index.tsx#L146

Comment (wley3337) [reply src/redux/registration/registration.actions.ts:111]: @geekingreen awesome! I'll add this to my branch and update the API calls and pushes. Not sure how we want to capture this pattern but I'll try and write something up in the README

File: src/utils/customHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Router } from 'react-router-dom';
+import { createBrowserHistory } from 'history';
+import { ReactNode, useLayoutEffect, useState } from 'react';
+
+/**
+ * Custom history object and browser router so that we
+ * can use the history.push() directly from sagas
+ */
+const history = createBrowserHistory();
```
Comment (wley3337) [inline src/utils/customHistory.tsx:9]: Do we want to leave this exported? Wasn't sure if this would take priority for auto imports or not 🤷

**Review by geekingreen — APPROVED** [review]

**Pull request merged**